src/main/java/org/opentripplanner/middleware/persistence/TypedPersistence.java

```diff
@@ -150,6 +155,34 @@ public T getById(String id) {
         return mongoCollection.find(filter).limit(maximum).into(new ArrayList<>());
     }
 
+    /**
+     * Build a filter for querying Mongo based on userId and from/to dates.
+     */
+    public static Bson buildFilter(String userId, Date fromDate, Date toDate) {
+        Set<Bson> clauses = new HashSet<>();
+        if (userId == null) {
+            throw new IllegalArgumentException("userId is required to be non-null.");
+        }
+        // user id is required, so as a minimum return all entities for user.
+        clauses.add(eq("userId", userId));
+        // Get all entities created since the supplied "from date".
+        if (fromDate != null) {
+            clauses.add(gte("dateCreated", fromDate));
+        }
+        // Get all entities created until the supplied "to date".
+        if (toDate != null) {
+            clauses.add(lte("dateCreated", toDate));
+        }
+        return Filters.and(clauses);
+    }
+
+    /**
+     * Build a filter for querying Mongo based on userId.
+     */
+    public static Bson buildFilter(String userId) {
```

Comment:
Please rename to something more descriptive like `filterByUserId`.

Reply:
Updated.